src/core/src/device/dummy.rs

```diff
@@ -0,0 +1,50 @@
+#![allow(missing_docs)]
+use device::{Device, Resources, Capabilities, SubmitInfo};
+use device::command::{GenericCommandBuffer};
+
+pub struct DummyDevice {
+    capabilities: Capabilities
+}
+
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+pub enum DummyResources{}
+
+pub type Buffer         = u32;
```

Comment:
No need for these types, really. Everything should be accessed by interested parties as `Resources::XXX` associated types.